### PR TITLE
Handle duplicate enums across services

### LINKED ISSUE #2767: Apollo Federation - Duplicate scalar in multiple services
- I run into an error when using a scalar like `DateTime` in two different microservices, and merging them together with Apollo Federation. Here is a snippet of the error being thrown ```UnhandledPromiseRejectionWarning: GraphQLSchemaValidationError: There can be only one type named "DateTime".```

- Apollo Federation

- `UnhandledPromiseRejectionWarning: GraphQLSchemaValidationError: There can be only one type named "DateTime"..`

### Microservice Foo
```
export const typeDefsFoo: DocumentNode = gql`
  scalar DateTime
  
  extend type Query {
    foo(id: ID!): Sensor!
  }

  type Foo {
    time: DateTime!
  }
```

### Microservice Bar
```
export const typeDefsBar: DocumentNode = gql`
  scalar DateTime
  
  extend type Query {
    bar(id: ID!): Bar!
  }

  type Bar {
    time: DateTime!
  }
`;
```

### PR DESCRIPTION
**Pull request by JakeDawkins**
1. Removes existing validations for duplicate type names for Scalars and Enums (we expect duplicate definitions)
2. Add new SDL validation rules for matching enums and scalars to allow for duplicates across services, but not _within_ a service.
3. Add pre-composition validation for duplicate enum _values_ within a single service

Resolves #2767

### CODE REVIEW

File: packages/apollo-federation/src/composition/validate/sdl/__tests__/matchingEnums.test.ts [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+        `,
+        name: 'serviceB',
+      },
+    ];
+
+    const definitionsDocument = createDefinitionsDocumentForServices(
+      serviceList,
+    );
+    const errors = validateSDL(definitionsDocument, schema, [matchingEnums]);
+    expect(errors).toMatchInlineSnapshot(`Array []`);
```
Comment (jbaxleyiii) [inline packages/apollo-federation/src/composition/validate/sdl/__tests__/matchingEnums.test.ts]: This is a total `nit` but I find this weird vs errors.length === 0

File: packages/apollo-federation/src/composition/validate/sdl/__tests__/matchingEnums.test.ts [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+      },
+    ];
+
+    const definitionsDocument = createDefinitionsDocumentForServices(
+      serviceList,
+    );
+    const errors = validateSDL(definitionsDocument, schema, [matchingEnums]);
+    expect(errors).toMatchInlineSnapshot(`
+      Array [
+        [GraphQLError: [serviceA] ProductType -> ProductType is an enum in [serviceA, serviceC], but not in [serviceB]],
```
Comment (jbaxleyiii) [inline packages/apollo-federation/src/composition/validate/sdl/__tests__/matchingEnums.test.ts]: This error could be clearer (the functionality is great)! Can we tell what kind of type it is in `serviceB` in this case? I.e. `ProductType is an enum in [serviceA, serviceC], but is an object type in [serviceB]`

File: packages/apollo-federation/src/composition/validate/sdl/index.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+export {
+  UniqueTypeNamesWithoutEnumsOrScalars,
+} from './uniqueTypeNamesWithoutEnumsOrScalars';
+export { matchingEnums } from './matchingEnums';
```
Comment (jbaxleyiii) [inline packages/apollo-federation/src/composition/validate/sdl/index.ts]: Let;s use the same PascalCase as above

File: packages/apollo-federation/src/composition/validate/sdl/__tests__/matchingEnums.test.ts (hunk 2 above)
Comment (JakeDawkins) [reply packages/apollo-federation/src/composition/validate/sdl/__tests__/matchingEnums.test.ts]: I'm not sure. What if there were multiple services with `ProductType` and one was an object type, and the other was a Scalar?

**Pull request merged**